From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 13 Apr 2022 01:42:14 +0200
Subject: monitor-manager: Ensure monitors settings after backend has been
 updated

The monitors settings such as the privacy screen property is propagated
to the monitors via kms updates, however during initialization and
on monitors changes, we end up clearing the pending KMS updates because
such settings are added to the queue before the backend has fully
initialized the monitors, and this may lead to discarding all the
pending updates, including the one we've just planned.

To avoid this, move settings applications after we've both initialized
the backend and notified it about changes.

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-control-center/+bug/1966178
Origin: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2372
---
 src/backends/meta-monitor-manager.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/backends/meta-monitor-manager.c b/src/backends/meta-monitor-manager.c
index 83fe107..67361ad 100644
--- a/src/backends/meta-monitor-manager.c
+++ b/src/backends/meta-monitor-manager.c
@@ -3561,10 +3561,10 @@ meta_monitor_manager_rebuild (MetaMonitorManager *manager,
 
   meta_monitor_manager_update_logical_state (manager, config);
 
-  ensure_monitors_settings (manager);
-
   meta_monitor_manager_notify_monitors_changed (manager);
 
+  ensure_monitors_settings (manager);
+
   g_list_free_full (old_logical_monitors, g_object_unref);
 }
 
@@ -3840,6 +3840,8 @@ meta_monitor_manager_post_init (MetaMonitorManager *manager)
   ClutterBackend *clutter_backend;
   ClutterSeat *seat;
 
+  ensure_monitors_settings (manager);
+
   clutter_backend = meta_backend_get_clutter_backend (manager->backend);
   seat = clutter_backend_get_default_seat (clutter_backend);
 
